web_chatter_right/static/src/js/chatter_right.js

```diff
@@ -0,0 +1,30 @@
+odoo.define('web_chatter_right.Chatter', function (require) {
+  'use strict';
+  /*
+   * Theme Chatter Right
+   * GNU Public License
+   * Alexandre Díaz <dev@redneboa.es>
+   */
+
+
+  var FormRenderer = require('web.FormRenderer');
+
+  FormRenderer.include({
+
+      _updateView: function ($newContent) {
+          this._super($newContent);
+          var $chatter = this.$el.find('.oe_chatter');
+          if (this.has_sheet && $chatter.length !== 0) {
+              if ($chatter.css('visibility') === 'hidden' || $chatter.css('display') === 'none') {
+                  this.$el.find('.o_form_sheet_bg').removeClass("o_form_sheet_chatter_right");
```

Comment:
Possibly you could do all of this overriding the XML template instead, and it would be more clear.

Also, users should be able to choose these settings.